lib/fizzy/instantiate.hpp

```diff
@@ -21,23 +21,47 @@ namespace fizzy
 struct ExecutionResult;
 struct Instance;
 
+using ExecuteFunctionPtr = ExecutionResult (*)(void* context, Instance&, const Value*, int depth);
+using DeleteContextFunctionPtr = void (*)(void* context) noexcept;
+
 /// Function representing WebAssembly or host function execution.
-using execute_function = std::function<ExecutionResult(Instance&, const Value*, int depth)>;
+class ExecuteFunction
+{
+public:
+    ExecuteFunction() = default;
+
+    template <class F>
+    ExecuteFunction(F f, void* context, DeleteContextFunctionPtr context_deleter)
+      : m_function{f}, m_context{context, context_deleter}
+    {}
+
+    template <class F>
+    ExecuteFunction(F f) : m_function{f}, m_context{nullptr, [](void*) noexcept {}}
+    {}
+
+    ExecuteFunction(Instance& instance, FuncIdx func_idx);
+
+    ExecutionResult operator()(Instance& instance, const Value* args, int depth);
```

Comment:
Wouldn't this be also `noexcept`? Or `m_context.get` is not.

Reply:
It  will be `noexcept`.
I'm thinking next to try extracting minimal needed changes from this PR and applying `noexcept` change. It will look much simpler than this PR, but will have extra dynamic allocations, whiche "does not look good" according to @chfast